### ISSUE TITLE
Parallel Runner & AutoScaling Issues

### ISSUE THREAD

**Issue by JavierGOrdonnez** [ITISFoundation/osparc-issues#1816]
**Long Story Short**
Using the Parallel Runner to run 50 instantiations of a large computational template, I see unexpected behaviours in the autoscaling. Private cluster is enabled. Also the parallel runner fails to keep 10 jobs running simultaneously. And the timeout fails - it waits until the job is done but then kills the worker?

**Expected Behavior**
10x 8xlarge machines spawn.

**Actual behaviour**
Just 5 of those spawn, and 2x machines are also created (maybe for the smaller, accesory services in the pipeline? But the pipeline is fully linear).

This screenshot is after 10hours of running, with 10 parallel jobs (no batching). I would expect 1-2h per job, thus 5-10h total runtime. However, there is still significant computation to get done.
![image](https://github.com/user-attachments/assets/af0c0772-9167-4925-b8a1-cc017f89d26d)
![image](https://github.com/user-attachments/assets/1af101c3-99da-43f4-88a3-5161ad8b8555)
![image](https://github.com/user-attachments/assets/f4f10e31-b619-4afe-b130-bd8aff7648e9)

**Comment by JavierGOrdonnez** [comment]
TLDR of the issues:
- @sanderegg  why no 10 8xlarge machines spawn?
- @wvangeit the timeout seems to be functioning wrong - kills the job AFTER it runs forever, and then doesnt spawn a new one.

These issues can be tackled after the break, no hurry. Will make this run work one way or the other over the weekend, and we can tackle this later.

**Comment by JavierGOrdonnez** [comment]
Btw this made the full run go to waste (eventually there werent any runners left) - and thus the associated large-machine costs. On the other hand, I believe this identifies the issue we were having a few weeks back :)
![image](https://github.com/user-attachments/assets/f238d512-3d97-4869-8064-ae6f3c4e4edd)

**Comment by JavierGOrdonnez** [comment]
Yes, I believe we have completely shifted towards the functions API now. Therefore, we can close this case (for now 😉)
